src/main/java/org/grpc/service/EmployeeGrpcImpl.java

```diff
@@ -0,0 +1,211 @@
+package org.grpc.service;
+
+import com.google.rpc.Code;
+import com.google.rpc.Status;
+import io.grpc.protobuf.StatusProto;
+import io.grpc.stub.StreamObserver;
+import org.grpc.service.entities.Employee;
+import org.grpc.service.entities.Shift;
+import org.grpc.service.repositories.EmployeeRepository;
+import org.grpc.service.repositories.ShiftRepository;
+import quickshift.grpc.service.*;
+import quickshift.grpc.service.Boolean;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EmployeeGrpcImpl extends EmployeeGrpc.EmployeeImplBase {
+    private ShiftRepository shiftRepository;
+    private EmployeeRepository employeeRepository;
+
+    public EmployeeGrpcImpl(ShiftRepository shiftRepository, EmployeeRepository employeeRepository) {
+        this.employeeRepository = employeeRepository;
+        this.shiftRepository = shiftRepository;
+    }
+
+    @Override
+    public void addSingleEmployee(NewEmployeeDTO request, StreamObserver<EmployeeDTO> responseObserver) {
+        if (employeeRepository.existsEmployeeByWorkingNumber(request.getWorkingNumber())) {
+            Status status = Status.newBuilder()
+                    .setCode(Code.ALREADY_EXISTS_VALUE)
+                    .setMessage("An employee with this working number already exists!")
+                    .build();
+            responseObserver.onError(StatusProto.toStatusRuntimeException(status));
+            return;
+        }
+
+        Employee newEmployee = new Employee(
+                request.getFirstName(),
+                request.getLastName(),
+                request.getWorkingNumber(),
+                request.getPassword(),
+                request.getEmail());
+
+        Employee savedEmployee = employeeRepository.save(newEmployee);
+        responseObserver.onNext(convertEmployeeToEmployeeDTO(savedEmployee, shiftRepository));
+        responseObserver.onCompleted();
+    }
+
+    @Override
+    public void getSingleEmployeeById(Id request, StreamObserver<EmployeeDTO> responseObserver) {
```

Comment:
What is StreamObserver?

Reply:
It's a class required by GRPC in java, it's like the client is the observer? I think...

Reply:
(at least that's what the client would expect for the method to return)